src/mlpack/methods/softmax_regression/softmax_regression_impl.hpp

```diff
@@ -14,47 +14,59 @@ namespace mlpack {
 namespace regression {
 
 template<template<typename> class OptimizerType>
+SoftmaxRegression<OptimizerType>::
+SoftmaxRegression(const size_t inputSize,
+                  const size_t numClasses,
+                  const bool fitIntercept) :
+  inputSize{inputSize},
+  numClasses{numClasses},
+  lambda{0.0001},
+  fitIntercept{fitIntercept}
+{
+  SoftmaxRegressionFunction regressor(arma::mat(), 1,
+                                      inputSize, numClasses,
+                                      lambda, fitIntercept);
+  parameters = regressor.GetInitialPoint();
+}
+
+template<template<typename> class OptimizerType>
+SoftmaxRegression<OptimizerType>::
+SoftmaxRegression(const std::string &fileName,
+                  const std::string& name)
+{
+  data::Load(fileName, name, *this, true);
+}
+
+template<template<typename> class OptimizerType>
 SoftmaxRegression<OptimizerType>::SoftmaxRegression(const arma::mat& data,
                                                     const arma::vec& labels,
                                                     const size_t inputSize,
                                                     const size_t numClasses,
                                                     const double lambda,
                                                     const bool fitIntercept) :
-    inputSize(inputSize),
-    numClasses(numClasses),
-    lambda(lambda),
-    fitIntercept(fitIntercept)
+  inputSize{inputSize},
+  numClasses{numClasses},
+  lambda{lambda},
+  fitIntercept{fitIntercept}
 {
   SoftmaxRegressionFunction regressor(data, labels, inputSize, numClasses,
                                       lambda, fitIntercept);
   OptimizerType<SoftmaxRegressionFunction> optimizer(regressor);
 
   parameters = regressor.GetInitialPoint();
-
-  // Train the model.
-  Timer::Start("softmax_regression_optimization");
-  const double out = optimizer.Optimize(parameters);
-  Timer::Stop("softmax_regression_optimization");
-
-  Log::Info << "SoftmaxRegression::SoftmaxRegression(): final objective of "
-      << "trained model is " << out << "." << std::endl;
+  Train(optimizer);
 }
 
 template<template<typename> class OptimizerType>
 SoftmaxRegression<OptimizerType>::SoftmaxRegression(
-    OptimizerType<SoftmaxRegressionFunction>& optimizer) :
-    parameters(optimizer.Function().GetInitialPoint()),
-    inputSize(optimizer.Function().InputSize()),
-    numClasses(optimizer.Function().NumClasses()),
-    lambda(optimizer.Function().Lambda())
+  OptimizerType<SoftmaxRegressionFunction>& optimizer) :
+  parameters(optimizer.Function().GetInitialPoint()),
+  inputSize{optimizer.Function().InputSize()},
+  numClasses{optimizer.Function().NumClasses()},
+  lambda{optimizer.Function().Lambda()},
+  fitIntercept{optimizer.Function().FitIntercept()}
```

Comment:
This should be four spaces in from the previous line; I'll fix this when I merge it in.  (It happens a couple other places too.)  I'm just pointing it out here so that you know. :)